### PR TITLE
Add comment in istio.deps

### PR DESCRIPTION
**Pull request by chxchx**
Sister PR  https://github.com/istio/test-infra/pull/709

### CODE REVIEW

**Comment by ZackButcher** [comment]
It's probably worth testing to see if things break if we include a regular old javascript comment (both line comment `//` and multi-line comment `/**/`) outside of the structure itself, rather than having to add the field as a hack.

**Comment by chxchx** [comment]
@ZackButcher Yeah I tried that but the json parser does not like it. The bigger issue is that when the robot serializes the updated Dependency struct in memory back to this file, the comments are gone, so having the comment as one of the field might be the only way to go.

File: istio.deps [hunk 1]
```diff
@@ -1,5 +1,6 @@
 [
 	{
+		"_comment": "This file is auto updated by a robot executing `https://github.com/istio/test-infra/blob/master/prow/test-infra-update-deps.sh`.",
```
Comment (ldemailly) [inline istio.deps]: :tada:

does the bot deal with vendor yet ?

Comment (chxchx) [reply istio.deps]: No fix for vendor yet and the bot is still disabled. krishna will be working on it although all of us are pretty occupied with other work. Still, having this comment at least explains what this file is for (and I got this question quite a few times)

**Pull request merged**